src/frontend/src/sol/services/sol-listener.services.ts

```diff
@@ -0,0 +1,61 @@
+import { balancesStore } from '$lib/stores/balances.store';
+import { i18n } from '$lib/stores/i18n.store';
+import { toastsError } from '$lib/stores/toasts.store';
+import type { TokenId } from '$lib/types/token';
+import type { SolPostMessageDataResponseWallet } from '$sol/types/sol-post-message';
+import { nonNullish } from '@dfinity/utils';
+import { BigNumber } from '@ethersproject/bignumber';
+import { get } from 'svelte/store';
+
+export const syncWallet = ({
+	data,
+	tokenId
+}: {
+	data: SolPostMessageDataResponseWallet;
+	tokenId: TokenId;
+}) => {
+	const {
+		wallet: {
+			balance: { certified, data: balance }
+		}
+	} = data;
+
+	if (nonNullish(balance)) {
+		balancesStore.set({
+			tokenId,
+			data: {
+				data: BigNumber.from(balance),
+				certified
+			}
+		});
+	} else {
+		balancesStore.reset(tokenId);
+	}
+
+	// Transaction handling will be added later
```

Comment:
Something to consider - after sol wallet worker supports transactions, maybe we can combine the btc and sol listeners into one single service as they seem to be completely identical. Transaction stores and error text from the function below can be passed as params. It obviously only one of many places where networks things can be combined, but in this case it seems to be very low hanging fruit which can spare 150 lines of code for us 🙂

Reply:
Yes, totally agree. Also it is planned to migrate eth to the worker approach, which then could benefit from this shared implementation.

Reply:
Note that having it separated has advantages too, notably a clear separation of concerns and the assurance that if one network fails, it does not impact the other—this without the need to handle the error.